examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -0,0 +1,211 @@
+        ax.set(xlabel='Model', title=title, xticks=x,
+               xticklabels=["Dropped", "One Hot", "Ordinal", "Native"],
+               ylim=y_limit)
+    fig.suptitle(figure_title)
+
+
+plot_results("Gradient Boosting on Adult Census")
+
+# %%
+# We see that the model with one-hot-encoded data is by far the slowest. This
+# is to be expected, since one-hot-encoding creates one additional feature per
+# category value (for each categorical feature), and thus more split points
+# need to be considered during fitting. The native handling of categorical
+# features is slightly slower than treating categories as ordered quantities
+# ('Ordinal'), since native handling requires :ref:`sorting categories
+# <categorical_support_gbdt>`.
+#
+# In terms of prediction performance, dropping the categorical features leads
+# to poorer performance. The three models that use categorical features have
+# comparable error rates, with a slight edge for the native handling.
+#
+# In general, one can expect poorer predictions from one-hot-encoded data,
+# especially when the the trees depths or the number of nodes are limited: with
+# one-hot-encoded data, one needs more split points (i.e. more depth) in order
+# to recover an equivalent split that could be obtained in one single split
+# point with native handling. This is also true when categories are treated as
+# ordinal quantities: if categories are `A..F` and the best split is `ACF -
+# BDE` the one-hot-encoder model will need 3 split points (one per category in
+# the left node), and the ordinal non-native model will need 4 splits: 1 split
+# to isolate `A`, 1 split to isolate `F`, and 2 splits to isolate `C` from
+# `BCDE`.
+
+# %%
+# Under-fitting variant
+# ----------------------
+#
+# In practice, how strongly the model performances differ will depend on the
+# dataset and on the flexibility of the trees.
+#
+# To see this, let us re-run the same analysis with under-fitting models where
+# we artificially limit the total number of splits by both limitting the number
+# of trees and the depth of each tree.
+
+for pipe in (hist_dropped, hist_one_hot, hist_ordinal, hist_native):
+    pipe.set_params(histgradientboostingregressor__max_depth=3,
+                    histgradientboostingregressor__max_iter=15)
+
+dropped_result = cross_validate(hist_dropped, X, y, cv=3, scoring=scoring)
+one_hot_result = cross_validate(hist_one_hot, X, y, cv=3, scoring=scoring)
+ordinal_result = cross_validate(hist_ordinal, X, y, cv=3, scoring=scoring)
+native_result = cross_validate(hist_native, X, y, cv=3, scoring=scoring)
+
+plot_results("Gradient Boosting on Adult Census (few and small trees)")
+
+plt.show()
+
+# %%
+# The results for these under-fitting models confirm our previous intuition:
+# the native category handling strategy fairs the best when the splitting
+# budget is constrained. The two other strategies (one-hot and ordinal
+# encoding) lead to error values comparable to the baseline model that just
```

Comment:
Since we still use the OE in the native handling maybe this will be less ambiguous:

```suggestion
# budget is constrained. The two other strategies (one-hot encoding and
# treating categories as ordinal values) lead to error values comparable
# to the baseline model that just
```